### DISCUSSION TITLE
How to run a sidecar with args from rust?

### DISCUSSION THREAD

**Discussion by jourdain** [tauri-apps/tauri#5379] [Q&A] [answered]
The following is working but I wanted to dynamically inject args and I'm not sure how to edit that part of the code.

```rust
let (mut rx, _) = Command::new_sidecar("server")
        .expect("failed to create sidecar")
        .spawn()
        .expect("Failed to spawn server");
```

Also in the config, I've added

```json
[...]
"tauri": {
    "allowlist": {
      "shell": {
        "sidecar": true,
        "scope": [
          {
            "name": "bin/server",
            "sidecar": true,
            "args": true
          }
        ]
      }
    },
[...]
```

Any guidance will be greatly appreciated.

Thanks,

Seb

**Comment by simonhyll** [comment, marked as answer]
Should just be to call `.args(["first","second"])` before `.spawn()`

https://docs.rs/tauri/1.1.1/tauri/api/process/struct.Command.html#method.args

**Reply by jourdain** [reply]
Thanks I've initially tried it right after the `.new_sidecar()` but couldn't understand the error.

**Comment by Merovex** [comment]
```
let (mut rx, mut child) = Command::new_sidecar("hello-world")
                .expect("failed to create `my-sidecar` binary command")
                .args([target])
                .spawn()
                .expect("Failed to spawn sidecar");
```

For those looking for a bit more detail. Simonhyll's answer was spot on. Thanks!
